test/integration/targets/azure_rm_networkinterface/tasks/main.yml

```diff
@@ -495,3 +495,127 @@
     name: "pip{{ rpfx }}"
     resource_group: '{{ resource_group }}'
     state: absent
+
+- name: create a network security group
+  azure_rm_securitygroup:
+    name: "nsg{{ rpfx }}"
+    resource_group: '{{ resource_group }}'
+
+- name: create a NIC with an existing NSG
+  azure_rm_networkinterface:
+      resource_group: "{{ resource_group }}"
+      name: "tn{{ rpfx }}nsg"
+      virtual_network: "{{ vn.state.id }}"
+      subnet: "tn{{ rpfx }}"
+      security_group:
+        name: "nsg{{ rpfx }}"
+        resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that:
+      - output.changed
+      - output.state.network_security_group.name == 'nsg{{ rpfx }}'
+      
+
+- name: create a NIC with an existing NSG (idempotent)
+  azure_rm_networkinterface:
+      resource_group: "{{ resource_group }}"
+      name: "tn{{ rpfx }}nsg"
+      virtual_network: "{{ vn.state.id }}"
+      subnet: "tn{{ rpfx }}"
+      security_group: "nsg{{ rpfx }}"
+  register: output
+
+- assert:
+    that:
+      - not output.changed
+      - output.state.network_security_group.name == 'nsg{{ rpfx }}'
+
+- name: create a NIC to add an existing NSG to
+  azure_rm_networkinterface:
+      resource_group: "{{ resource_group }}"
+      name: "tn{{ rpfx }}nsgadd"
+      virtual_network: "{{ vn.state.id }}"
+      subnet: "tn{{ rpfx }}"
+  register: output
+
+- assert:
+    that:
+      - output.changed
+
+- name: add an existing NSG to an existing NIC
+  azure_rm_networkinterface:
+    resource_group: "{{ resource_group }}"
+    name: "tn{{ rpfx }}nsgadd"
+    virtual_network: "{{ vn.state.id }}"
+    subnet: "tn{{ rpfx }}"
+    security_group: "nsg{{ rpfx }}"
+  register: output
+
+- assert:
+    that:
+      - output.changed
+      - output.state.network_security_group.name == 'nsg{{ rpfx }}'
+
+- name: add an existing NSG to an existing NIC (idempotent)
```

Comment:
I think from Line498 to Line 575 are duplicate with https://github.com/ansible/ansible/blob/e6ffc4f89a27853e2ce983bc51a982b5a138d1bd/test/integration/targets/azure_rm_networkinterface/tasks/main.yml#L75-L99

We can add the disassociate NSG test after "Create NIC using virtual_network resource_group parameter (idempotent)" to avoid redundant resource created during integration test.

Reply:
@marc-sensenich Thanks for your contribution, do you finished changed?